src/components/cards/Card/Card.tsx

```diff
@@ -0,0 +1,62 @@
+import React, { MouseEvent, useMemo } from "react";
+
+import * as S from "./card.style";
+
+export type CardInfo = {
+  cardName: string;
+  cardNumber: string;
+  cardOwnerName: string;
+  expireDate: string;
+};
+
+export interface CardProps {
+  size: "big" | "small";
+  className?: string;
+  color?: string;
+  cardInfo: CardInfo;
+  onClick?: (e: MouseEvent<HTMLDivElement>) => void;
+}
+
+export default function Card({
+  size,
+  color = "#94dacd",
+  cardInfo,
+  className,
+}: CardProps) {
+  const { cardName, cardNumber, cardOwnerName, expireDate } = cardInfo;
+
+  const cardNumberWithMasking = useMemo(
+    () =>
+      cardNumber
+        .split("-")
+        .map((value, idx) => {
+          if (idx < 2) {
+            return value;
+          } else {
+            return value.replaceAll(/[0-9]/g, "•");
+          }
```

Comment:
```suggestion
          const MASKING_NEDDED_MIN_INDEX = 3;
          
          function checkMaskingNeedFIeld() {
              return index >= MASKING_NEDDED_MIN_INDEX;
          }
          
          const isMaskingNeeded = checkMaskingNeedFIeld(index);
        
          return isMaskingNeeded ? value.replaceAll(/[0-9]/g, "•") : value;
```
조금 더 길어지긴 하지만 이렇게 작성해두는게 좋아보여요 👀

Reply:
좋은 의견 감사합니다. 🙇‍♂️
한 가지 궁금한 점이 `map` 내부에 함수를 선언하게 되면,  반복이 될 때 마다 함수와 변수가 선언되고,
외부의 `index`를 클로저 형식으로 참조하게 되는 것 같은데, 이런 경우 문제가 되지는 않는지 질문드립니다. 🙇‍♂️

Reply:
외부에 선언하는게 맞다고 생각해요 👀 
suggestion으로 예시를 보여주다보려니 위처럼 작성을 했었어요 😓